languages/cpp/src/Secrets.cpp

```diff
@@ -0,0 +1,149 @@
+#include "Secrets.h"
+#include <nlohmann/json.hpp>
+#include <boost/uuid/uuid_io.hpp>
+#include <boost/uuid/string_generator.hpp> 
+#include <boost/uuid/uuid_io.hpp>
+
+Secrets::Secrets(CommandRunner* commandRunner) : commandRunner(commandRunner) {}
+
+auto secretsDeserializer = [](const std::string& response) -> ResponseForSecretResponse {
+    nlohmann::json jsonResponse = nlohmann::json::parse(response);
+    ResponseForSecretResponse secretResponse;
+    Bitwarden::Sdk::from_json(jsonResponse, secretResponse);
+    return secretResponse;
+};
+
+auto deleteSecretsDeserializer = [](const std::string& response) -> ResponseForSecretsDeleteResponse {
+    nlohmann::json jsonResponse = nlohmann::json::parse(response);
+    ResponseForSecretsDeleteResponse deleteSecretsResponse;
+    Bitwarden::Sdk::from_json(jsonResponse, deleteSecretsResponse);
+    return deleteSecretsResponse;
+};
+
+auto secretListDeserializer = [](const std::string& response) -> ResponseForSecretIdentifiersResponse {
+    nlohmann::json jsonResponse = nlohmann::json::parse(response);
+    ResponseForSecretIdentifiersResponse listResponse;
+    Bitwarden::Sdk::from_json(jsonResponse, listResponse);
+    return listResponse;
+};
+
+ResponseForSecretResponse Secrets::get(const boost::uuids::uuid& id) {
+    Command command;
+    SecretsCommand secretsCommand;
+    SecretGetRequest secretGetRequest;
+
+    std::string idStr = boost::uuids::to_string(id);
+    secretGetRequest.set_id(idStr);
+
+    secretsCommand.set_get(secretGetRequest);
+    command.set_secrets(secretsCommand);
+
+    try {
+        return commandRunner->runCommand<ResponseForSecretResponse>(command, secretsDeserializer);
+    } catch (const std::exception& ex) {
+        std::cerr << "Error in getSecret: " << ex.what() << std::endl;
+        throw ex;
+    }
+}
```

Comment:
Now that we verify that `runCommand` is a success and throw an exception when not, we can remove the unnecessary `ResponseFor*` wrapper types from the return types of the functions and just return the internal data, here's one example:

```suggestion
SecretResponse Secrets::get(const boost::uuids::uuid& id) {
    Command command;
    SecretsCommand secretsCommand;
    SecretGetRequest secretGetRequest;

    std::string idStr = boost::uuids::to_string(id);
    secretGetRequest.set_id(idStr);

    secretsCommand.set_get(secretGetRequest);
    command.set_secrets(secretsCommand);

    try {
        return commandRunner->runCommand<ResponseForSecretResponse>(command, secretsDeserializer).get_data().get();
    } catch (const std::exception& ex) {
        std::cerr << "Error in getSecret: " << ex.what() << std::endl;
        throw ex;
    }
}
```

We just had to append `.get_data().get()` to the runCommand call and remove the ResponseFor from the return type. Only thing missing would be to update the header files and the functions in BitwardenClient

Reply:
But if we use `return commandRunner->runCommand<SecretResponse>(command, secretsDeserializer)` we do not need to append .get_data().get(), right?

Just to change the type of the response to `SecretResponse`, also in `secretsDeserializer` we need to change type to `SecretResponse`.

Reply:
Actually yes, that is a better approach. I will change this and come back to you.